DataProcessor/DataGen.cpp

```diff
@@ -28,36 +24,34 @@ void DataGen::getData(QByteArray &data, std::vector<std::string> &names, std::ve
                 addFloatToArray(data,(float)batteryFunc(timeArg));
             } else if(names[i] == "accelerator") {
                 addFloatToArray(data,std::fmod(timeArg,5)/*(float)rand()/((RAND_MAX+1u)/5)*/);
+            } else if(names[i] == "speed") {
+                addFloatToArray(data,(float)speedFunc(timeArg));
             } else {
                 addFloatToArray(data,(float)rand()/((RAND_MAX+1u)/100));
             }
         } else if(types[i] == "uint8") {
-            if(names[i] == "speed") {
-                dataToByteArray(data,(uint8_t)speedFunc(timeArg));
-            } else {
-                dataToByteArray(data,(uint8_t)fmod(rand(),200));
-            }
+            dataToByteArray(data,(uint8_t)fmod(rand(),200));
         } else if(types[i] == "uint16") {
             dataToByteArray(data,(uint16_t)fmod(rand(),200));
         } else if(types[i] == "bool") {
             // For shutdown circuit inputs, any triggerred error will stay triggered for 3 seconds after the most recent error is triggered
             // When a new error is triggerred, the countdown will restart from 3 seconds for all currently triggered errors
-            if((names[i] == "battery_eStop") || (names[i] == "driver_eStop") || (names[i] == "external_eStop") ||
-               (names[i] == "imd_status") || (names[i] == "door") || (names[i] == "mcu_check")) {
+            if(names[i] == "door") {
                 // NC/preferred true shutdown circuit inputs
                 std::size_t errPos = errors.find(names[i]);
-                bool error = (rand()%200+1 >= 2) && !((errStartTime > time(NULL) - 3) && (errPos != std::string::npos));
+                bool error = (rand()%700+1 >= 2) && !((errStartTime > time(NULL) - 3) && (errPos != std::string::npos));
```

Comment:
Let's remove magic numbers or at least make a function to tell me what is going on here